File: 006/Euler 6.py
__author__ = 'DT'

def euler6():
    sum_of_squares = sum([x**2 for x in range(101)])
    square_of_sum = sum([y for y in range(101)])**2
    return (square_of_sum - sum_of_squares)

print euler6()

print ((sum([y for y in range(101)])**2) - (sum([x**2 for x in range(101)])))

File: 006/Euler 6.js
(function euler6() {

  var each = function(n, cb){
    // iterates through all numbers 1..n and performs callback
    for (var i = 1; i <= n; i++){
      cb(i);
    }
  };

  var sumSquares = function(n){
    var sum = 0;
    each(n, function(i){
      sum += i;
    });
    return sum * sum;
  };

  var sqauresSum = function(n){
    var sum = 0;
    each(n, function(i){
      sum += i * i;
    });
    return sum;
  };

  var result = sumSquares(100) - sqauresSum(100);
  return result;

})();
